guides/theming.md

```diff
@@ -1,311 +1,359 @@
+));
+```
 
-### Multiple themes
+#### Applying a theme to components
 
-You can create multiple themes for your application by including the `angular-material-theme` mixin
-multiple times, where each inclusion is gated by an additional CSS class.
+The `core-theme` Sass mixin emits prerequisite styles for common features used by multiple
+components. This mixin must be included once per theme.
 
-Remember to only ever include the `@mat-core` mixin once; it should not be included for each
-theme.
+Each Angular Material component has a "color" mixin that emits the component's color styles and
+a "typography" mixin that emits the component's typography styles.
 
-#### Example of defining multiple themes:
+Additionally, each component has a "theme" mixin that emits styles for both color and typography if
+they are specified in the theme defined by `define-light-theme` or `define-dark-theme`. Apply the
+styles for each of the components used in your application by including each of their theme Sass
+mixins.
 
 ```scss
-@import '~@angular/material/theming';
-// Plus imports for other components in your app.
-
-// Include the common styles for Angular Material. We include this here so that you only
-// have to load a single css file for Angular Material in your app.
-// **Be sure that you only ever include this mixin once!**
-@include mat-core();
-
-// Define the default theme (same as the example above).
-$candy-app-primary: mat-palette($mat-indigo);
-$candy-app-accent:  mat-palette($mat-pink, A200, A100, A400);
-$candy-app-theme:   mat-light-theme((
-  color: (
-    primary: $candy-app-primary,
-    accent: $candy-app-accent,
-  ),
-  typography: mat-typography-config()
+@use '~@angular/material' as mat;
+
+$my-primary: mat.define-palette(mat.$indigo-palette, 500);
+$my-accent: mat.define-palette(mat.$pink-palette, A200, A100, A400);
+
+$my-theme: mat.define-light-theme((
+ color: (
+   primary: $my-primary,
+   accent: $my-accent,
+ )
 ));
 
-// Include the default theme styles (color and default density)
-@include angular-material-theme($candy-app-theme);
+// Emit theme-dependent styles for common features used across multiple components. 
+@include mat.core-theme($my-theme);
+
+// Emit styles for MatButton based on `$my-theme`. Because the configuration
+// passed to `define-light-theme` omits typography, `button-theme` will not
+// emit any typography styles.
+@include mat.button-theme($my-theme);
+
+// Include the theme mixins for other components you use here.
+```
+
+As an alternative to listing every component that your application uses, Angular Material offers
+Sass mixins that includes styles for all components in the library: `all-component-colors`,
+`all-component-typographies`, and `all-component-themes`. These mixins behave the same as individual
+component mixins, except they emit styles for `core-theme` and _all_ 35+ components in Angular
+Material. Unless your application uses every single component, this will produce unecessary CSS.
```

Comment:
What is the override order here if I specified multiple?
`core-theme` > `all-component-themes` > `all-component-colors` || `all-component-typographies` > `<component-name>-theme` ?

Reply:
It's standard CSS specificity rules- the last one would win, unless you nest the `@include` inside a more specific rule declaration. 

To clarify, though:
* `core-theme` includes only styles for common stuff used across multiple components, such as ripples. 
* `all-component-themes` just individually calls every single component's `-theme` mixin
* Each component's theme mixin calls through to that component's color and typography (and density) mixins

None of these mixins add additional rule declarations, so the CSS you end up with is exactly the same, e.g.:
```scss
@include mat.button-theme(...);
// is exactly* equivalent to
@include mat.button-color(...);
@include mat.button-typography(...);

// which both emit the exact same button styles as
@include mat.all-component-themes(...);

// but `all-component-themes` includes all the *other* components too

// * assuming your theme includes a color and typography config
```